webpack.config.js

```diff
@@ -1,38 +1,68 @@
-const HtmlWebPackPlugin = require("html-webpack-plugin");
+const HtmlWebPackPlugin = require('html-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const webpack = require('webpack');
 
-module.exports = {
+module.exports = (env = {}) => {
+
+  const {mode = 'development'} = env;
+
+  const isProd = mode === 'production';
+  const isDev = mode === 'development';
+
+  const getStyleLoaders = () => {
+    return [
+      isProd? MiniCssExtractPlugin.loader : 'style-loader',
+      'css-loader'
+    ];
+  };
+
+  const getPlugins = () => {
+    const plugins = [
+      new HtmlWebPackPlugin({
+        template: './src/index.html',
+        filename: './index.html'
+      })
+    ];
+
+    if(isProd) {
+      plugins.push(new MiniCssExtractPlugin({
+        filename: 'main-[hash:8].css'
+      }));
+    }
+
+    if(isDev) {
+      plugins.push(new webpack.SourceMapDevToolPlugin({
+        test: /\.js$/,
+        filename: '[name].js.map',
+      }));
+    }
+
+    return plugins;
+  };
+
+  return {
+    mode: isProd ? 'production' : isDev && 'development',
```

Comment:
instead of using ternary operator you could just pass env.mode or it has some purpose?

Reply:
Yup, the idea was that if someone breaks the "start" script or runs build from terminal without specifying the environment parameter, we will simply run with mode=false, so wepback will use no env-specific default optimizations